libvips/colour/colour.c

```diff
@@ -245,40 +242,43 @@ vips_colour_build(VipsObject *object)
 
 	if (vips_image_generate(out,
 			vips_start_many, vips_colour_gen, vips_stop_many,
-			in, colour)) {
-		g_object_unref(out);
+			in, colour))
 		return -1;
-	}
 
 	/* Reattach higher bands, if necessary. If we have more than one input
 	 * image, just use the first extra bands.
 	 */
-	for (i = 0; i < colour->n; i++)
+	for (int i = 0; i < colour->n; i++)
 		if (extra_bands[i]) {
-			VipsImage *t1, *t2;
+			VipsImage **t = (VipsImage **) vips_object_local_array(object, 5);
```

Comment:
nit:
```suggestion
			VipsImage **t = (VipsImage **) vips_object_local_array(object, 3);
```
(although I understand that oversizing this array can be useful for future adjustments, feel free to disregard this)